feature/savings/src/commonMain/kotlin/com/mifos/feature/savings/savingsAccountv2/pages/PreviewPage.kt

```diff
@@ -10,26 +10,185 @@
+import androidclient.feature.savings.generated.resources.feature_savings_yes
+import androidclient.feature.savings.generated.resources.step_charges
+import androidclient.feature.savings.generated.resources.step_charges_active
+import androidclient.feature.savings.generated.resources.step_charges_view
+import androidclient.feature.savings.generated.resources.step_details
+import androidclient.feature.savings.generated.resources.step_terms
+import androidclient.feature.savings.generated.resources.step_terms_apply_withdrawal_fee
+import androidclient.feature.savings.generated.resources.step_terms_decimal_places
+import androidclient.feature.savings.generated.resources.step_terms_is_allowed_overdraft
+import androidclient.feature.savings.generated.resources.step_terms_lock_in_period
+import androidclient.feature.savings.generated.resources.step_terms_min_opening_balance
+import androidclient.feature.savings.generated.resources.step_terms_minimum_balance
 import androidx.compose.foundation.layout.Column
 import androidx.compose.foundation.layout.Spacer
+import androidx.compose.foundation.layout.fillMaxSize
 import androidx.compose.foundation.layout.height
-import androidx.compose.material3.Button
+import androidx.compose.foundation.layout.padding
+import androidx.compose.foundation.lazy.LazyColumn
 import androidx.compose.material3.Text
 import androidx.compose.runtime.Composable
-import androidx.compose.ui.Alignment
 import androidx.compose.ui.Modifier
-import androidx.compose.ui.unit.dp
+import com.mifos.core.designsystem.theme.DesignToken
+import com.mifos.core.designsystem.theme.MifosTypography
+import com.mifos.core.ui.components.MifosGeneralCard
+import com.mifos.core.ui.components.MifosRowWithTextAndButton
+import com.mifos.core.ui.components.MifosTwoButtonRow
+import com.mifos.feature.savings.savingsAccountv2.SavingsAccountAction
+import com.mifos.feature.savings.savingsAccountv2.SavingsAccountState
 import org.jetbrains.compose.resources.stringResource
 
 @Composable
-fun PreviewPage(onNext: () -> Unit) {
-    Column(horizontalAlignment = Alignment.CenterHorizontally) {
-        Text(stringResource(Res.string.step_preview))
-        Spacer(Modifier.height(8.dp))
-        Button(onClick = onNext) {
-            Text(stringResource(Res.string.feature_savings_submit))
+fun PreviewPage(
+    state: SavingsAccountState,
+    onAction: (SavingsAccountAction) -> Unit,
+    modifier: Modifier = Modifier,
+) {
+    val previewDetailsMap = mapOf(
+        stringResource(Res.string.feature_savings_product_name) + " : " to state.savingProductOptions[state.savingsProductSelected].name,
+        stringResource(Res.string.feature_savings_field_officer) + " : " to state.fieldOfficerOptions[state.fieldOfficerIndex].displayName,
+        stringResource(Res.string.feature_savings_submission_date) + " : " to state.submissionDate,
+        stringResource(Res.string.feature_savings_external_id) + " : " to state.externalId,
```

Comment:
Don’t hardcode `:` in strings put formatting in your UI component (or use string resources with placeholders), not in the keys. If MifosGeneralCard requires the colon, let it add it; if not possible today, at least keep labels clean and add the colon once when rendering.